hack/verify-golint.sh

```diff
@@ -46,88 +46,90 @@ array_contains () {
 # Check that the file is in alphabetical order
 failure_file="${KUBE_ROOT}/hack/.golint_failures"
 if ! diff -u "${failure_file}" <(LC_ALL=C sort "${failure_file}"); then
-	{
-		echo
-		echo "hack/.golint_failures is not in alphabetical order. Please sort it:"
-		echo
-		echo "  LC_ALL=C sort -o hack/.golint_failures hack/.golint_failures"
-		echo
-	} >&2
-	false
+  {
+    echo
+    echo "hack/.golint_failures is not in alphabetical order. Please sort it:"
+    echo
+    echo "  LC_ALL=C sort -o hack/.golint_failures hack/.golint_failures"
+    echo
+  } >&2
+  false
 fi
 
 export IFS=$'\n'
 # NOTE: when "go list -e ./..." is run within GOPATH, it turns the k8s.io/kubernetes
 # as the prefix, however if we run it outside it returns the full path of the file
 # with a leading underscore. We'll need to support both scenarios for all_packages.
 all_packages=(
-	$(go list -e ./... | egrep -v "/(third_party|vendor|staging/src/k8s.io/client-go/pkg|generated|clientset_generated)" | sed -e 's|^k8s.io/kubernetes/||' -e "s|^_${KUBE_ROOT}/\?||")
+  $(go list -e ./... | egrep -v "/(third_party|vendor|staging/src/k8s.io/client-go/pkg|generated|clientset_generated)" | sed -e 's|^k8s.io/kubernetes/||' -e "s|^_${KUBE_ROOT}/\?||")
 )
 failing_packages=(
-	$(cat $failure_file)
+  $(cat $failure_file)
 )
 unset IFS
 errors=()
 not_failing=()
 for p in "${all_packages[@]}"; do
-	# Run golint on package/*.go file explicitly to validate all go files
-	# and not just the ones for the current platform.
-	# Packages with a corresponding foo_test package will make golint fail
-	# with a useless error. Just ignore that, see golang/lint#68.
-	failedLint=$(golint "$p"/*.go 2>/dev/null)
-	array_contains "$p" "${failing_packages[@]}" && in_failing=$? || in_failing=$?
-	if [[ -n "${failedLint}" ]] && [[ "${in_failing}" -ne "0" ]]; then
-		errors+=( "${failedLint}" )
-	fi
-	if [[ -z "${failedLint}" ]] && [[ "${in_failing}" -eq "0" ]]; then
-		not_failing+=( $p )
-	fi
+  # Run golint on package/*.go file explicitly to validate all go files
+  # and not just the ones for the current platform. This also will ensure that
+  # _test files are linted.
+  # Generated files are ignored, and each file is passed through golint
+  # individually, as if one file in the package contains a fatal error, golint
```

Comment:
Can we add some more details on why a package might contain a fatal error? Something like: `as if one file in the package contains a fatal error due to the presence of a corresponding foo_test package`.

The `foo_test` line in the script helped me find the problem in a second. :)